src/System.Drawing.Common/src/System/Drawing/Font.Windows.cs

```diff
@@ -311,17 +311,21 @@ public static unsafe Font FromLogFont(object lf, IntPtr hdc)
             // Now that we know the marshalled size is the same as LOGFONT, copy in the data
             logFont = new SafeNativeMethods.LOGFONT();
 
-            if (!type.IsValueType)
-            {
-                // Only works with non value types
-                Marshal.StructureToPtr(lf, new IntPtr(&logFont), fDeleteOld: false);
-            }
-            else
+            try
             {
                 GCHandle handle = GCHandle.Alloc(lf, GCHandleType.Pinned);
                 Buffer.MemoryCopy((byte*)handle.AddrOfPinnedObject(), &logFont, nativeSize, nativeSize);
                 handle.Free();
             }
+            catch (ArgumentException)
+            {
+                // If the type wasn't blittable it won't be able to be pinned with GCHandle. Try
+                // to use the marshaller to copy the "native" representation instead.
+                //
+                // This happens for classes or structs that have objects non fixed arrays in them,
```

Comment:
"that have objects non fixed arrays"?  I'm having trouble parsing that... is it missing a word?

Reply:
Apparently I don't write well when I'm tired. 😂